app/src/org/commcare/dalvik/activities/EntitySelectActivity.java

```diff
@@ -681,19 +696,59 @@ public boolean onCreateOptionsMenu(Menu menu) {
             menu.add(0, MENU_MAP, MENU_MAP, Localization.get("select.menu.map")).setIcon(
                     android.R.drawable.ic_menu_mapmode);
         }
-        Action action = shortSelect.getCustomAction();
+        final Action action = shortSelect.getCustomAction();
         if(action != null) {
             ViewUtil.addDisplayToMenu(this, menu, MENU_ACTION, action.getDisplay());
         }
 
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB) {
+            final MenuInflater inflater = getMenuInflater();
+            inflater.inflate(R.menu.activity_report_problem, menu);
+
+            searchView =
+                    (SearchView)menu.findItem(R.id.search_action_bar).getActionView();
+            if (searchView != null) {
+                searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+                    @Override
+                    public boolean onQueryTextSubmit(final String query) {
+                        return true;
+                    }
+
+                    @Override
+                    public boolean onQueryTextChange(final String newText) {
+                        if (adapter != null) {
+                            adapter.applyFilter(newText);
+                        }
+                        return false;
+                    }
+                });
+                final int[] searchViewStyle = AndroidUtil.getThemeColorIDs(this, new int[]{R.attr.searchbox_action_bar_color});
+                final int id = searchView.getContext()
+                        .getResources()
+                        .getIdentifier("android:id/search_src_text", null, null);
+                final TextView textView = (TextView) searchView.findViewById(id);
+                textView.setTextColor(searchViewStyle[0]);
+            }
+
+            final View bottomSearchWidget = findViewById(R.id.searchfooter);
+            if(bottomSearchWidget != null) bottomSearchWidget.setVisibility(View.GONE);

```

Comment:
Don't we want to not-do-this if the build version is < honeycomb?

Also: I think we shouldn't check for null here. If the bottom search view isn't there because we changed the file, we should delete this code.